scripts/typescript_checks.py

```diff
@@ -57,13 +811,123 @@ def validate_compiled_js_dir() -> None:
+
+        with utils.open_file(STRICT_TSCONFIG_FILEPATH, 'w') as f:
+            json.dump(strict_ts_config, f, indent=2, sort_keys=True)
+            f.write('\n')
+
+        # Compile tsconfig-strict.json with files those are neither strictly
+        # typed nor present in TS_STRICT_EXCLUDE_PATHS. All those files
+        # present inside include property.
+        os.environ['PATH'] = '%s/bin:' % common.NODE_PATH + os.environ['PATH']
+        validate_compiled_js_dir()
+
+        if os.path.exists(COMPILED_JS_DIR):
+            shutil.rmtree(COMPILED_JS_DIR)
+
+        cmd = ['./node_modules/typescript/bin/tsc', '--project', config_path]
+        process = subprocess.Popen(
+            cmd, stdout=subprocess.PIPE, encoding='utf-8')
+
+        # The value of `process.stdout` should not be None since we passed
+        # the `stdout=subprocess.PIPE` argument to `Popen`.
+        assert process.stdout is not None
+        error_messages = list(iter(process.stdout.readline, ''))
+
+        # Update tsconfig-strict.json and set to its intial "include" state
+        # example "include": ["core", "extensions", "typings"].
+        with utils.open_file(STRICT_TSCONFIG_FILEPATH, 'r') as f:
+            strict_ts_config = yaml.safe_load(f)
+            strict_ts_config['include'] = prefixes
+
+        with utils.open_file(STRICT_TSCONFIG_FILEPATH, 'w') as f:
+            json.dump(strict_ts_config, f, indent=2, sort_keys=True)
+            f.write('\n')
+
+        if error_messages:
+            print('\n' + '\n'.join(error_messages))
+            print(
+                str(len([x for x in error_messages if x.startswith(prefixes)]))
+                + ' Errors found during compilation.\n')
+            sys.exit(1)
+        else:
+            print('Compilation successful!')
+
+        yield
+    finally:
+        with utils.open_file(STRICT_TSCONFIG_FILEPATH, 'r') as f:
+            strict_ts_config = yaml.safe_load(f)
+            strict_ts_config['include'] = prefixes
+
+        with utils.open_file(STRICT_TSCONFIG_FILEPATH, 'w') as f:
+            json.dump(strict_ts_config, f, indent=2, sort_keys=True)
+            f.write('\n')
+
+
 def compile_and_check_typescript(config_path: str) -> None:
     """Compiles typescript files and checks the compilation errors.
 
     Args:
         config_path: str. The config that should be used to run the typescript
             checks.
     """
+    # Set strict TS config include property to ["core", "extensions", "typings"]
+    # This make sure to restore include property to its original value after the
+    # checks get aborted mid-way.
+    with utils.open_file(STRICT_TSCONFIG_FILEPATH, 'r') as f:
+        strict_ts_config = yaml.safe_load(f)
+        strict_ts_config['include'] = (
+            ['core', 'extensions', 'typings'])
+
+    with utils.open_file(STRICT_TSCONFIG_FILEPATH, 'w') as f:
+        json.dump(strict_ts_config, f, indent=2, sort_keys=True)
+        f.write('\n')
```

Comment:
I thought this was already being handled by the try-except above. Why do we need to do this again?

Reply:
if running script is close in-between(assume we closes the terminal), we can't restore include property. So add this for correct execution of next compilation.

Reply:
It sounds like this code is to handle the case where the script is aborted part-way through such that no cleanup code is allowed to run (e.g. with `kill -9`). I don't think you need to worry about that case. We just want to make sure that if a user uses something like CTRL-C to stop the script but then lets the script end gracefully (e.g. so `finally` blocks still run), we clean up the tsconfig file.

Reply:
I only worry about next run of command. If Include property not restore after kill of terminal.

Is it true that I don't have to be concerned about that case?

Reply:
No need now, make separate strict ts-config, see [this](https://github.com/oppia/oppia/pull/15578#discussion_r900016964) discussion.